Src/Newtonsoft.Json/Serialization/JsonArrayContract.cs

```diff
@@ -207,11 +212,7 @@ public JsonArrayContract(Type underlyingType)
                 _parameterizedConstructor = CollectionUtils.ResolveEnumerableCollectionConstructor(underlyingType, CollectionItemType);
 
 #if HAVE_FSHARP_TYPES
-                if (!HasParameterizedCreatorInternal && underlyingType.Name == FSharpUtils.FSharpListTypeName)
```

Comment:
You will still need this when using FSharp.Core 4.4.1.0 and before.  FSharp.Core only impemented IEnumerable up to version 4.4.1.0 inclusive